swarm/network/stream/stream.go

```diff
@@ -619,19 +625,60 @@ func (p *Peer) HandleMsg(ctx context.Context, msg interface{}) error {
 		return p.handleUnsubscribeMsg(msg)
 
 	case *OfferedHashesMsg:
-		return p.handleOfferedHashesMsg(ctx, msg)
+		go func() {
+			err := p.handleOfferedHashesMsg(ctx, msg)
+			if err != nil {
```

Comment:
The error should be logged.

Since most of the messages are creating a goroutine, would it be simpler to have a goroutine with the whole switch statement in it and remove the repetitive code?

Reply:
Error is logged now, you're probably looks at an outdated diff.

The repetitive code is just a log line and a p.Drop(), I don't have preference if it should be in an external function or not.

Reply:
Thanks for the log line. Maybe others can express their opinions about repetition in this function, it is working, but I do not find it nice.

Reply:
The mental load that this puts to the reader is close to 0. It really doesn't matter. We should focus on other parts of Swarm, but this.

Reply:
You are right about the mental load, but my opinion is that this function can be structured to be easier to maintain. I think that every part of the swarm is important equally if we want to have a better code. But it is not a big deal, as I said, it works.

Reply:
Handling incoming messages asynchronously should probably be part of the `p2p/protocols` abstraction, which is not really in scope for this PR.

In light of our `guidelines` and `standards`, I think that this PR is increasing the code quality from the previous state of the code, because:
- a lot of dead code is being removed
- a bug is being solved

Therefore I suggest we address the repetition of the `go log error drop` flow, since we have a few other bugs that are not yet integrated to the codebase.

Reply:
i am good with leaving these functions as is right now for clarity.
if we keep the error (and actually use it) in `peer.Drop` we could drop the log because the error will be automatically logged on the peer drop